src/cript/nodes/util/__init__.py

```diff
@@ -175,7 +192,26 @@ def __call__(self, node_str: Union[dict, str]) -> dict:
 
         no serialization is needed in this case and just needs to be converted from str to dict
 
-        if the node field is present then continue and convert the JSON node into a Python object
+        if the node field is present, then continue and convert the JSON node into a Python object
+
+        Parameters
+        ----------
+        node_str : Union[dict, str]
+            The JSON representation of a node or a regular dictionary.
+
+        Returns
+        -------
+        dict
+            A Python dictionary representing the deserialized node or the input dictionary itself.
```

Comment:
It returns a regular dictionary if the input JSON or input dict is not a node.

If it is a node, it returns a the appropriate node object.
For example, if you give it a JSON of Material node, it will return an object of type `cript.Material`.

Reply:
updated it